src/MRR/Simulator/mymath.py

```diff
@@ -23,12 +23,15 @@ def _gcd(x: float, y: float, n: float = 0) -> float:
     """
         x > y
     """
+    n = 0
+    while y != 0 and not is_zero(x, y) and n < 10:
+        x, y = y, mod(x, y)
+        n += 1
+
     if y == 0:
         return x
-    elif is_zero(x, y) or n > 10:
+    elif is_zero(x, y):
```

Comment:
no good

Reply:
this problem was fixed. bad ending condition

Reply:
ok